Packetery/Checkout/Block/Adminhtml/Order/View/Tab/Packets.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Packetery\Checkout\Block\Adminhtml\Order\View\Tab;
+
+use Magento\Backend\Block\Widget\Container;
+use Magento\Backend\Block\Widget\Context;
+use Magento\Backend\Block\Widget\Tab\TabInterface;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Phrase;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\OrderRepositoryInterface;
+
+class Packets extends Container implements TabInterface
+{
+    private const NAME = 'Packets';
+    private const PACKETERY_CARRIER_CODE = 'packetery';
+
+    public function __construct(
+        Context $context,
+        private readonly OrderRepositoryInterface $orderRepository,
+        private readonly RequestInterface $request,
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+    }
+
+    protected function _construct(): void
+    {
+        parent::_construct();
+        $this->setTemplate('Packetery_Checkout::packets.phtml');
+    }
+
+    public function getTabLabel(): Phrase
+    {
+        return __(self::NAME);
+    }
+
+    public function getTabTitle(): Phrase
+    {
+        return $this->getTabLabel();
+    }
+
+    public function canShowTab(): bool
+    {
+        $order = $this->getOrder();
+        if ($order === null) {
+            return false;
+        }
+
+        if ($this->isPacketeryShippingGroup($order->getShippingMethod())) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function isHidden(): bool
+    {
+        return !$this->canShowTab();
+    }
+
+    private function getOrder(): ?OrderInterface
+    {
+        $orderId = (int) $this->request->getParam('order_id');
+
+        return $this->orderRepository->get($orderId);
```

Comment:
dotaz: je potřeba tahat Magento order z databáze, když jde o záložku v detailu objednávky ? nelze získat magento objednávku v constructoru ?

Reply:
Magento order muzu tahat jeste pres registry(), ale tato funkce jiz byla deprikovana. Jedina moznost mi tedy dava smysl tahat to z DB.

Reply:
opraveno - už se objednávka netahá z DB